src/World.cc

```diff
@@ -245,48 +250,51 @@ Errors World::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
 
   for (const auto &ifaceModelPair : this->dataPtr->interfaceModels)
   {
-    frameNames.insert(ifaceModelPair.second->Name());
+    implicitFrameNames.insert(ifaceModelPair.second->Name());
   }
 
-  // Load all the physics.
-  if (_sdf->HasElement("physics"))
+  for (auto elem = _sdf->GetFirstElement(); elem; elem = elem->GetNextElement())
   {
-    this->dataPtr->physics.clear();
-    Errors physicsLoadErrors = loadUniqueRepeated<Physics>(_sdf, "physics",
-        this->dataPtr->physics);
-    errors.insert(errors.end(), physicsLoadErrors.begin(),
-        physicsLoadErrors.end());
+    const std::string elementName = elem->GetName();
+    if (elementName == "model")
+    {
+      auto model = loadSingle<Model>(errors, elem, _config);
+      if (!recordUniqueName(nestedModelNames, elementName, model.Name()))
+      {
+        continue;
+      }
+      implicitFrameNames.insert(model.Name());
+      this->dataPtr->models.push_back(std::move(model));
+    }
+    else if (elementName == "joint")
+    {
+      auto joint = loadSingle<Joint>(errors, elem);
+      if (!recordUniqueName(jointNames, elementName, joint.Name()))
+      {
+        continue;
+      }
+      this->dataPtr->joints.push_back(std::move(joint));
+    }
+    else if (elementName == "frame")
+    {
+      auto frame = loadSingle<Frame>(errors, elem);
+      if (!recordUniqueName(explicitFrameNames, elementName, frame.Name()))
+      {
+        continue;
+      }
+      this->dataPtr->frames.push_back(std::move(frame));
+    }
   }
 
-  // Load all the actors.
-  Errors actorLoadErrors = loadUniqueRepeated<Actor>(_sdf, "actor",
-      this->dataPtr->actors);
-  errors.insert(errors.end(), actorLoadErrors.begin(), actorLoadErrors.end());
-
-  // Load all the joints.
-  Errors jointLoadErrors = loadUniqueRepeated<Joint>(_sdf, "joint",
-      this->dataPtr->joints);
-  errors.insert(errors.end(), jointLoadErrors.begin(), jointLoadErrors.end());
-
-  // Load all the lights.
-  Errors lightLoadErrors = loadUniqueRepeated<Light>(_sdf, "light",
-      this->dataPtr->lights);
-  errors.insert(errors.end(), lightLoadErrors.begin(), lightLoadErrors.end());
-
-  // Load all the frames.
-  Errors frameLoadErrors = loadUniqueRepeated<Frame>(_sdf, "frame",
-      this->dataPtr->frames);
-  errors.insert(errors.end(), frameLoadErrors.begin(), frameLoadErrors.end());
-
   // Check frames for name collisions and modify and warn if so.
```

Comment:
not something new here, but I just noticed that this logic block always sets `WARNING` and doesn't check `if (sdfVersion < gz::math::SemanticVersion(1, 7))` and set `DUPLICATE_NAME` for newer versions, like is done in Model.cc

Reply:
Good catch. Should we address it in a separate PR?

Reply:
yeah, that's fine

Reply:
opened a draft PR in https://github.com/gazebosim/sdformat/pull/1311